test/unit/spec/common/jobCommChannel-spec.js

```diff
@@ -129,24 +132,225 @@ define([
+            // send three pings over the bus
+            // init the comm channel after the second ping
+            // the first two pings should generate an error and be stored
+            // the third ping should be processed successfully
+            return new Promise((resolve) => {
+                spyOn(comm, 'sendCommMessage').and.callFake(async (...args) => {
+                    if (!args.length) {
+                        if (comm.comm) {
+                            spyOn(comm.comm, 'send');
+                        }
+                        return comm.sendCommMessage.and.originalFn.call(comm, ...args);
+                    }
+                    const { pingId } = args[1];
+                    if (pingId < 2) {
+                        // original function should throw an error
+                        await expectAsync(
+                            comm.sendCommMessage.and.originalFn.call(comm, ...args)
+                        ).toBeRejectedWithError(
+                            'ERROR sending comm message: ' + comm.ERROR_COMM_CHANNEL_NOT_INIT
+                        );
+                    } else {
+                        // call the original function
+                        await comm.sendCommMessage.and.originalFn.call(comm, ...args);
+                    }
+                    // resolve the promise on receiving the second ping
+                    if (pingId === 1) {
+                        resolve();
+                    }
+                });
+                // emit the first two pings
+                [0, 1].forEach((ix) => {
+                    testBus.emit(...pingArgs[ix]);
+                });
+            }).then(() => {
+                // first two pings will be stored
+                const expectedQueue = pingArgs
+                    .map((arg) => {
+                        return {
+                            msgType: arg[0],
+                            msgData: arg[1],
+                        };
+                    })
+                    .slice(0, 2);
+                expect(comm.messageQueue).toEqual(expectedQueue);
+                // init the comm channel and send the last ping
+                return comm.initCommChannel().then(() => {
+                    expect(comm.comm).not.toBeNull();
+                    return new Promise((resolve) => {
+                        // resolve the promise when we see comm.comm.send(...)
+                        spyOn(comm.comm, 'send').and.callFake(() => {
+                            resolve();
+                        });
+                        // emit the last ping
+                        testBus.emit(...pingArgs[2]);
+                    }).then(() => {
+                        expect(comm.sendCommMessage).toHaveBeenCalledTimes(pingArgs.length + 1);
+                        expect(comm.sendCommMessage.calls.allArgs()).toEqual([
+                            pingArgs[0],
+                            pingArgs[1],
+                            [],
```

Comment:
this is from when `comm.sendCommMessage()` gets called at the end of the `comm.initCommChannel` method.